Logger

trecouvr opened the pull request:
related to #17 

I had to add a new dependency : SlimExtras, the default logger of Slim is too basic.

trecouvr committed:
create \Payutc\Log

trecouvr committed:
Update Paybox to use \Payutc\Log

trecouvr committed:
Update MADMIN to use \Payutc\Log

trecouvr committed:
Remove class/Log.class.php

trecouvr committed:
Fix bugs in Log class

Remove require config.inc.php
Fix second parameters of array_key_exists in __callstatic

trecouvr committed:
Add unittests for logger

mattgu74 commented:
J'ai testé, déjà ça ne casse rien, c'est pas trop mal ;) (au niveau de casper/paybox/MADMIN, enfin la ou ça a changé des choses quoi).

Par contre, tu as du oublier de fournir le config.inc.dist.php, du coup je ne sais pas ou et comment configurer pour avoir les logs, donc je ne peux pas dire si l'objet principal de la pull request fonctionne... Donc en attendant que tu nous en dises plus on peux pas merge.

trecouvr committed:
Use SlimExtras LogWriter

The default LogWriter of Slim is too basic. This LogWriter allow
you to specify a file and prefix the log message with the level
and the date. This can be configured of course.
I also put a default configuration in config.inc.dist.php

mattgu74 commented on config.inc.dist.php:
```diff
@@ -68,8 +68,16 @@
+    'log.level' => \Slim\Log::DEBUG,
+    'log.enabled' => true,
+    'log.writer' => new \Slim\Extras\Log\DateTimeFileWriter(array(
+	'path' => __DIR__.'/logs',
```
ça ne m'a pas l'air très bien indenté tout ça ;)

trecouvr replied:
corrigé

mattgu74 commented:
J'ai testé donc premier point en faveur de l'acceptation : "ça ne casse rien" ;)

Par contre:
J'ai ajouté les nouvelles entrées dans ma config.
J'ai fais un composer update
J'ai créer un dossier logs (en chmod 777 pour être sur d'avoir le droit d'écrire dedans)
Et j'ai aucun log qui se crée quand je fais un rechargement...

Du coup ou vont les logs ?

trecouvr commented:
Les logs vont dans **DIR**.'/logs'. Je crois que le dossier doit déjà exister et pour sûr il doit être "ecrivable" par ton web user.

trecouvr committed:
Correct indentation

mattgu74 commented:
Bon ok je valide, c'est juste qu'on a des logs que sur les erreurs dans paybox lors de l'analyse du paybox retour... Et c'est plutot rare enfait.
Le seul log "fréquent", c'est en mettant un faux numéro de carte, en annulant un rechargement, une carte périmé...
Donc j'ai testé et il passe c'est bon, il a fini par s'écrire. ;)

Donc bref je valide la PR, mais ce serait bien qu'on ouvre un ticket "Ajouter des logs aux endroits critiques ou ça pourrait nous servir" 
Parce que actuellement il y'en a sur les tentatives de rechargement négatifs et sur les erreurs paybox c'est tout.
